### PR TITLE
javascriptlint#29 "which" is useless for finding command paths

### LINKED ISSUE #29: `which` is useless for finding command paths
I've found that `PY_EXEC=$(shell which python2.7)` is extremely terrible.

* which always exits 0
* which emits errors on stdout
* make will define `PY_EXEC` as an empty string even if there is no output.

This means that `PY_EXEC` will always be set to *something*, even if it's garbage, and later `ifndef PY_EXEC` will never evaluate.

A better option for macOS is `type -a --path` (because we need to detect possibly multiple pythons, and omit `/usr/bin/python`. For other systems `command -v` does a better job.

Instead of `ifndef PY_EXEC` we need to use `ifeq ($(PY_EXEC),)`.

### PR DESCRIPTION
**Pull request by bahamat**
Fixes #29

### CODE REVIEW

**Comment by bahamat** [comment]
Tested this by updating a repo on both macOS and SmartOS to this commit in the deps and ran `make check`. Both pass.

File: Makefile [hunk 1]
```diff
@@ -21,11 +21,11 @@ CPPFLAGS += -DNDEBUG -D_REENTRANT					\
 ifeq ($(BUILDOS),Darwin)
 #	As of macOS 12, we can't use the system python anymore, so take the
 #	first one that's not in /usr/bin.
-	PY_EXEC=$(shell which -a python2.7 | grep -v /usr/bin/python | head -1)
+	PY_EXEC=$(shell type -a --path python2.7 | grep -v /usr/bin/python | head -1)
 else
-	PY_EXEC=$(shell which python2.7)
+	PY_EXEC=$(shell /usr/bin/command -v python2.7)
```
Comment (danmcd) [inline Makefile:26]: THIS is the magic that works on both bash and ksh93.  Thank you, and thank @jperkin too!

**Review by danmcd — APPROVED** [review]
We'll have to rebuild the whole world, won't we?

**Comment by bahamat** [comment]
> We'll have to rebuild the whole world, won't we?

Very good question, but thankfully, no we won't need to rebuild the whole world.

The existing version works (accidentally) because everything has python2.7. This issue was discovered compiling on a system with only python2.6. I could have just down-rev'd this submodule (to a version before the magic detection) to work around it on systems with only python2.6, but it would have hit us again at some point in the future when switching to python3. But this fix should future-proof it so that when we switch to python3, it'll fall back gracefully to python2.7, python2, then python.

**Pull request merged**